lib/cocoapods/external_sources/abstract_external_source.rb

```diff
@@ -169,9 +169,11 @@ def store_podspec(sandbox, spec, json = false)
         rescue Pod::DSLError => e
           raise Informative, "Failed to load '#{name}' podspec: #{e.message}"
         end
+        defined_in_file = spec.defined_in_file
         spec.defined_in_file = nil
         validate_podspec(spec)
```

Comment:
perhaps changing `defined_in_file` should be done in the `validate_podspec` method?